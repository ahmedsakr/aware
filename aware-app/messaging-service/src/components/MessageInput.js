import React, {Component} from 'react';
import '../style/MessageInput.css';

class MessageInput extends Component {
  render() {
    return (
      <div id="message-input">
        <img id="insert-png" src={process.env.PUBLIC_URL + "/icons8-document-80.png"} alt="PDF"/>
        <input class="col-9" id="textfield" type="text" name="message" placeholder="Type here.."/>
        <input id="send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" />
      </div>
    );
  }
}

export default MessageInput;
