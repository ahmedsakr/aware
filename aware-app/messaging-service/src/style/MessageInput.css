#textfield {
  height: 35px;
  font-size: 24px;
  margin-left: 15px;
  margin-right: 15px;
}

#insert-png {
    vertical-align: middle;
    display: inline-block;
    height: 65px;
}

#send-img {
    vertical-align: middle;
    display: inline-block;
    height: 65px;
}

#message-input {
    text-align: left;
}