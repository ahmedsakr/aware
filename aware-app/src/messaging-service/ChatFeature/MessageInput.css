#message-input-textfield {
  font-size: 24px;
}

#message-input-insert-png, #message-input-send-img {
    height: 65px;
}

#message-input-send-img {
    vertical-align: middle;
}

#message-input {
    text-align: left;
}