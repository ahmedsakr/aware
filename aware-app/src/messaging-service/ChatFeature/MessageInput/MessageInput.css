#message-input-textfield {
  font-size: 24px;
}

#message-input-insert-png, #message-input-send-img {
    height: 65px;
}

#message-input-send-img {
    vertical-align: middle;
}

#message-input {
    background-color: #eaeff1;
    text-align: left;
}

form {
    flex: 0 1 auto;
}