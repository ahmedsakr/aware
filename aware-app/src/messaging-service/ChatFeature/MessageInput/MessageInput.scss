/*
 * Message Input styling
 *
 * Defines layout fundamentals for the Message Input Field.
 */

$message_input_background_color: #eaeff1;

#message-input-textfield {
    background-color: $message_input_background_color;
    border: none;
    height: 100%;
}

#message-input-send-img {
    vertical-align: middle;
}

#message-input {
    flex: 1 0 auto;
    background-color: $message_input_background_color;
    text-align: left;
}

form {
    flex: 0 1 auto;
    display: flex;
    flex-flow: column;
}