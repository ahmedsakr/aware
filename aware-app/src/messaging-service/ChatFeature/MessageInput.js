import React, {Component} from 'react';
import './MessageInput.css';
import {produceTimestamp} from '../../aware-utils.js'

class MessageInput extends Component {
  constructor() {
    super()
    this.state = {
       message: ''
    }
    //Need to bind 'this' to give access to keyword
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div id="message-input">
          <img id="insert-png" src={process.env.PUBLIC_URL + "/icons8-document-80.png"} alt="PDF"/>
          <input class="col-9" id="textfield" type="text" name="message" placeholder="Type here.." onChange={this.handleChange} value={this.state.message}/>
          <input id="send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" />
        </div>
      </form>
    );
  }

  handleChange(e) {
    this.setState({
      //Set state variable to input value
      message: e.target.value
    })
  }

  handleSubmit(e) {
    //Prevents page refresh when form submit
    e.preventDefault();
    // Build a message object to be sent
    const messageObj = {
      studentName: "Josh",
      text: this.state.message,
      timestamp: produceTimestamp(),
      avatar: "/icons8-user-80blue.png"
    };
    this.props.sendMessage(messageObj);
    //Reset state of input to erase previous message
    this.setState({
      message: ''
    })
  }
}

export default MessageInput;