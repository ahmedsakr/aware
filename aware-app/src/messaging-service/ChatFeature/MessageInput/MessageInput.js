import React, {Component} from 'react';
import './MessageInput.css';
import {produceTimestamp} from '../../../aware-utils.js'

class MessageInput extends Component {
  constructor() {
    super()
    this.state = {message: ''}

    //Need to bind 'this' to give access to keyword
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div id="message-input">
          <input class="col-11" id="message-input-textfield" type="text" name="message" placeholder="Type here.." onChange={this.handleChange} value={this.state.message} />
          <input class="col-1" id="message-input-send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" disabled={!this.state.message} />
        </div>
      </form>
    );
  }

  handleChange(e) {
    this.setState({
      //Set state variable to input value
      message: e.target.value
    })
  }

  handleSubmit(e) {
    //Prevents page refresh when form submit
    e.preventDefault();
    // Build a message object to be sent
    const messageObj = {
      studentName: this.props.name,
      text: this.state.message,
      timestamp: produceTimestamp(),
      avatar: "/" + this.props.name + "-pic.jpg"
    };
    this.props.sendMessage(messageObj);

    //Reset state of input to erase previous message
    this.setState({
      message: ''
    })
  }
}

export default MessageInput;
