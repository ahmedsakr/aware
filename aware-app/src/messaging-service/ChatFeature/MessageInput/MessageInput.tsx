import React from 'react';
import './MessageInput.css';
import {UserMessageContents} from '../ChatWindow/Message/Message'
import produceTimestamp from '../../../aware-utils'

type MessageInputProps = {
  sendMessage: (message: UserMessageContents) => void,
  name: string
};

type MessageInputState = {
  message: string
};

export default class MessageInput extends React.Component<MessageInputProps, MessageInputState> {
  constructor(props: MessageInputProps) {
    super(props)

    this.state = {
      message: ''
    };

    // Need to bind 'this' to give access to keyword
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({
      //Set state variable to input value
      message: event.target.value
    })
  }

  handleSubmit(event: React.FormEvent<HTMLFormElement>): void {
    //Prevents page refresh when form submit
    event.preventDefault();

    // Build a message object to be sent
    let messageObj: UserMessageContents = {
      username: this.props.name,
      content: this.state.message,
      timestamp: produceTimestamp(),
      avatar: "/" + this.props.name + "-pic.jpg"
    };

    this.props.sendMessage(messageObj);

    //Reset state of input to erase previous message
    this.setState({
      message: ''
    })
  }

  render(): JSX.Element {
    return (
      <form onSubmit={this.handleSubmit}>
        <div id="message-input">
          <input className="col-11" id="message-input-textfield" type="text" name="message" placeholder="Type here.." onChange={this.handleChange} value={this.state.message} />
          <input className="col-1" id="message-input-send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" disabled={!this.state.message} />
        </div>
      </form>
    );
  }
}
