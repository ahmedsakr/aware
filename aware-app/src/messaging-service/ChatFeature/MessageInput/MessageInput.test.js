import React from 'react';
import { shallow } from 'enzyme';

import MessageInput from './MessageInput';

describe('MessageInput', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<MessageInput debug />);
  
    expect(component).toMatchSnapshot();
  });
});