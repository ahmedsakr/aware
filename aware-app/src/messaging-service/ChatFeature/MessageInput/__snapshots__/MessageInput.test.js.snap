version https://git-lfs.github.com/spec/v1
oid sha256:07844ad6e8c22208d7620b337e9d7eb7163dd91eae648d391c55d3687fb57643
size 557
